### PR TITLE
add: resultのdenoテストの追加

### PR DESCRIPTION
**Pull request by kouphasi**
## やったこと
close: #1
`result.ts`にテストを追加

## 動作確認
```
Result module ...
  success ...
    should create a success result with the given value ... ok (1ms)
  success ... ok (1ms)
  failure ...
    should create a failure result with the given error ... ok (0ms)
  failure ... ok (0ms)
  isSuccess ...
    should return true for success result ... ok (0ms)
    should return false for failure result ... ok (0ms)
  isSuccess ... ok (0ms)
  isFailure ...
    should return true for failure result ... ok (0ms)
    should return false for success result ... ok (0ms)
  isFailure ... ok (0ms)
Result module ... ok (1ms)

ok | 1 passed (10 steps) | 0 failed (3ms)
```

### CODE REVIEW

File: result.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+describe("Result module", () => {
+  describe("success", () => {
+    it("should create a success result with the given value", () => {
+      const result = success(42);
+      assertEquals(result.data, 42);
+    });
+  });
+
+  describe("failure", () => {
+    it("should create a failure result with the given error", () => {
```
Comment (YuukiHayashi0510) [inline result.test.ts]: [must]
success同様にObject、null、undefinedのケースに加えて
カスタムエラー型（Errorを継承した型）のケース追加お願いします。

File: result.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  describe("isSuccess", () => {
+    it("should return true for success result", () => {
+      const result = success(42);
+      assertEquals(isSuccess(result), true);
+    });
+
+    it("should return false for failure result", () => {
+      const result = failure(new Error("test"));
+      assertEquals(isSuccess(result), false);
+    });
```
Comment (YuukiHayashi0510) [inline result.test.ts]: [imo]
テーブル駆動テストにしましょう

**Review by YuukiHayashi0510 — CHANGES_REQUESTED** [review]
コントリビュートありがとうございます！
コメントしたので、確認お願いします🙏

File: result.test.ts (hunk 1 above)
Comment (kouphasi) [reply result.test.ts]: こちらで対応しました
[722e1a6](https://github.com/YuukiHayashi0510/core.ts/pull/6/commits/722e1a6a1f909a3020f09d2e1b0fbbd8a22f2dd4)

File: result.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  describe("failure", () => {
+    it("should create a failure result with the given error", () => {
+      const error = new Error("test error");
+      const result = failure(error);
+      assertEquals(result.error, error);
+      assertEquals(result.error.message, "test error");
+    });
+  });
+
+  describe("isSuccess", () => {
```
[the remark this answers is not in the document]
Comment (kouphasi) [reply result.test.ts]: こちらで対応しました
[52157a8](https://github.com/YuukiHayashi0510/core.ts/pull/6/commits/52157a80b16cd6cf8083bf0da7b2c4e60194526d)

File: result.test.ts [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    describe("happy path", () => {
+      it("should create a failure result with the given error", () => {
+        const error = new Error("test error");
+        const result = failure(error);
+        assertEquals(result.error, error);
+        assertEquals(result.error.message, "test error");
+      });
+    });
+
+    describe("edge case testing", () => {
```
Comment (YuukiHayashi0510) [inline result.test.ts]: testing不要です
```suggestion
    describe("edge case", () => {
```

File: result.test.ts [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      ];
+
+      happyCases.forEach(({ name, result, expected, description }) => {
+        it(`should handle ${name} - ${description}`, () => {
+          assertEquals(isFailure(result), expected);
+          if (isFailure(result)) {
+            assertEquals("error" in result, true);
+          } else {
+            assertEquals("data" in result, true);
+          }
```
Comment (YuukiHayashi0510) [inline result.test.ts]: [imo]
シンプルに記述できそうです。
```suggestion
          assertEquals("error" in result, isFailure(result));
```

**Review by YuukiHayashi0510 — COMMENTED** [review]
ご対応いただいた、テーブル駆動テスト修正関連で追加コメントしています。
ご確認お願いします。

File: result.test.ts [hunk 6]
```diff
@@ -0,0 +1,146 @@
+import { describe, it } from "@std/testing/bdd";
+import { assertEquals } from "@std/assert";
+import { failure, isFailure, isSuccess, success } from "./result.ts";
+
+class CustomError extends Error {}
```
Comment (YuukiHayashi0510) [inline result.test.ts]: [must]
スコープが遠すぎるので、failure > edge case のスコープに移動してください

File: result.test.ts [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+          assertEquals(isSuccess(result), expected);
+          if (isSuccess(result)) {
+            assertEquals("data" in result, true);
+          } else {
+            assertEquals("error" in result, true);
+          }
+        });
+      });
+    });
+  });
```
[the remark this answers is not in the document]
Comment (YuukiHayashi0510) [reply result.test.ts]: 処理が重複しているので、分けなくて良いと思います。
対応するとして`description`が期待結果の文言になっているので、`description`変数名の修正の方かと思います。

[the remark this answers is not in the document]
Comment (kouphasi) [reply result.test.ts]: こちらで対応しました

[890cf92](https://github.com/YuukiHayashi0510/core.ts/pull/6/commits/890cf92d5b1c01b1fc5d8e5c55b6f0a6e0977bca)

[the remark this answers is not in the document]
Comment (YuukiHayashi0510) [reply result.test.ts]: detail：詳細ではなく、「期待結果」つまりwantやexpectなど適切な名称に変更してください

**Comment by YuukiHayashi0510** [comment]
雑めにコンフリクト解消しているので、フォーマット系とtypoらしき修正のみ対応お願いします。

**Comment by kouphasi** [comment]
@YuukiHayashi0510 
> 雑めにコンフリクト解消しているので、対応お願いします。

こちらでコンフリクトを解消しました
[2fcc5ce](https://github.com/YuukiHayashi0510/core.ts/pull/6/commits/2fcc5ce84614524bce0ab28536dae410fd8ca821)
